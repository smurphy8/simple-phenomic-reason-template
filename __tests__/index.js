// @flow

/* eslint-disable import/no-extraneous-dependencies */
import globby from "globby";

it("should build example correctly", () => {
  const testFolder = __dirname + "/../dist";
  const files = globby.sync("**/*", {
    cwd: testFolder,
    nodir: true,
    dot: true,
  });

  // should have html files
  const htmlFiles = files.filter(file => file.endsWith(".html"));
  expect(htmlFiles).toMatchSnapshot();

  const jsonApiFiles = files.filter(
    file => file.startsWith("phenomic") && file.endsWith(".json"),
  );
  // should have matching json files
  expect(jsonApiFiles).toMatchSnapshot();

  // should have assets
  const assetsBundlerFiles = files.filter(
    file =>
      !htmlFiles.includes(file) &&
      !jsonApiFiles.includes(file) &&
      file.startsWith("phenomic"),
  );
  expect(assetsBundlerFiles.length).toBe(1);
  expect(
    files.filter(
      file =>
        !htmlFiles.includes(file) &&
        !jsonApiFiles.includes(file) &&
        !assetsBundlerFiles.includes(file),
    ),
  ).toMatchSnapshot();
});
