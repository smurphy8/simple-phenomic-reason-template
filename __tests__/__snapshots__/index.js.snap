// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should build example correctly 1`] = `
Array [
  "404.html",
  "after/c2Vjb25kLXBvc3Q=/index.html",
  "after/dGhpcmQtcG9zdA==/index.html",
  "after/Zm91cnRoLXBvc3Q=/index.html",
  "after/ZmlmdGgtcG9zdA==/index.html",
  "after/Zmlyc3QtcG9zdA==/index.html",
  "blog/fifth-post/index.html",
  "blog/first-post/index.html",
  "blog/fourth-post/index.html",
  "blog/second-post/index.html",
  "blog/third-post/index.html",
  "index.html",
]
`;

exports[`should build example correctly 2`] = `
Array [
  "phenomic/content/posts/by-default/1/desc/date/limit-2.json",
  "phenomic/content/posts/by-default/1/desc/date/limit-2/after-c2Vjb25kLXBvc3Q=.json",
  "phenomic/content/posts/by-default/1/desc/date/limit-2/after-dGhpcmQtcG9zdA==.json",
  "phenomic/content/posts/by-default/1/desc/date/limit-2/after-Zm91cnRoLXBvc3Q=.json",
  "phenomic/content/posts/by-default/1/desc/date/limit-2/after-ZmlmdGgtcG9zdA==.json",
  "phenomic/content/posts/by-default/1/desc/date/limit-2/after-Zmlyc3QtcG9zdA==.json",
  "phenomic/content/posts/item/fifth-post.json",
  "phenomic/content/posts/item/first-post.json",
  "phenomic/content/posts/item/fourth-post.json",
  "phenomic/content/posts/item/second-post.json",
  "phenomic/content/posts/item/third-post.json",
]
`;

exports[`should build example correctly 3`] = `Array []`;
